### PR TITLE
chore(Java): Examples for Mutations

### PR DESCRIPTION
**Pull request by texastony**
_Issue #, if available:_

_Description of changes:_

_Squash/merge commit message, if applicable:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AwsCryptographicMaterialProviders/runtimes/java/build.gradle.kts [hunk 1]
```diff
@@ -83,6 +106,15 @@ dependencies {
...
     // https://mvnrepository.com/artifact/org.testng/testng
     testImplementation("org.testng:testng:7.5")
+
+    // Example Dependencies are marked as testImplementation
+    testImplementation("software.amazon.awssdk:arns")
+    testImplementation("software.amazon.awssdk:auth")
+    testImplementation("software.amazon.awssdk:sts")
+    testImplementation("software.amazon.awssdk:apache-client:2.19.0")
+    testAnnotationProcessor("org.projectlombok:lombok:1.18.30")
+    testImplementation("com.google.code.findbugs:jsr305:3.0.2")
```
Comment (josecorella) [inline AwsCryptographicMaterialProviders/runtimes/java/build.gradle.kts:116]: fix to better approach

File: AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystoreadmin/model/ApplyMutationOutput.java [hunk 2]
```diff
@@ -103,14 +103,11 @@ public ApplyMutationOutput build() {
...
-      ) {
-        throw new IllegalArgumentException(
-          "The size of `mutatedBranchKeyItems` must be greater than or equal to 1"
-        );
-      }
+      // if (Objects.nonNull(this.mutatedBranchKeyItems())) {
+      //   throw new IllegalArgumentException(
+      //     "The size of `mutatedBranchKeyItems` must be greater than or equal to 1"
+      //   );
+      // }
```
Comment (josecorella) [inline AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystoreadmin/model/ApplyMutationOutput.java:110]: remove pls

File: AwsCryptographicMaterialProviders/runtimes/java/src/examples/java/software/amazon/cryptography/example/hierarchy/MutationExample.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      System.out.println(
+        "ApplyLogs: " +
+        branchKeyId +
+        " items: \n" +
+        AdminProvider.mutatedItemsToString(applyOutput.mutatedBranchKeyItems())
+      );
+
+      if (result.continueMutation() != null) token = result.continueMutation();
+      if (result.completeMutation() != null) done = true;
+      if (limitLoop == 0) done = true;
```
[the remark this answers is not in the document]
Comment (texastony) [reply AwsCryptographicMaterialProviders/runtimes/java/src/examples/java/software/amazon/cryptography/example/hierarchy/MutationExample.java:91]: Resolved in 6b46a17c8

File: AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystoreadmin/model/ApplyMutationOutput.java (hunk 2 above)
Comment (texastony) [reply AwsCryptographicMaterialProviders/runtimes/java/src/main/smithy-generated/software/amazon/cryptography/keystoreadmin/model/ApplyMutationOutput.java:110]: Resolved awhile ago.

**Pull request merged**